From 88df19e864e90f738610b455706931476641b8d9 Mon Sep 17 00:00:00 2001
From: pershoot <28478728-pershoot@users.noreply.gitlab.com>
Date: Mon, 14 Sep 2026 11:50:00 -0400
Subject: [PATCH] kernel_patches: fs/susfs: Allow core to be built with no
 features

-Accommodate:
 -9/14/26: 'kernel: Fix NULL deref for susfs_get_enabled_features()'
           (upstream: susfs: 44a893a6)
---
 kernel_patches/fs/susfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel_patches/fs/susfs.c b/kernel_patches/fs/susfs.c
index 1e70f7f2..b55d9560 100644
--- a/kernel_patches/fs/susfs.c
+++ b/kernel_patches/fs/susfs.c
@@ -1219,7 +1219,7 @@ out_copy_to_user:
 }
 
 /* get susfs enabled features */
-static int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *copied_size, size_t bufsize) {
+static __maybe_unused int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *copied_size, size_t bufsize) {
 	size_t tmp_size = strlen(config_string);
 
 	*copied_size += tmp_size;
@@ -1234,7 +1234,7 @@ static int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *
 void susfs_get_enabled_features(void __user **user_info) {
 	struct st_susfs_enabled_features *info = (struct st_susfs_enabled_features *)kzalloc(sizeof(struct st_susfs_enabled_features), GFP_KERNEL);
 	char *buf_ptr = NULL;
-	size_t copied_size = 0;
+	__maybe_unused size_t copied_size = 0;
 	int err = 0;
 
 	if (!info) {
-- 
GitLab

